### DISCUSSION TITLE
How to document parameters defined implicitly?

### DISCUSSION THREAD

**Discussion by lfoppiano** [apiflask/apiflask#208] [Q&A] [answered]
Hi, 
  I've started using this library and I find it quite neat and well documented. 

I have an issue to correctly generate the documentation specs for a method whose parameters are defined implicitly: 

```
@bp.route("/records", methods=["GET"])
@output(Record(many=True))
def get_records_from_form_data():
    type = request.args.get('type', default="automatic", type=str)
    status = request.args.get('status', default="valid", type=str)
    publisher = request.args.get('publisher', default=None, type=str)
    year = request.args.get('year', default=None, type=str)
    start = request.args.get('start', default=0, type=int)
    limit = request.args.get('limit', default=-1, type=int)

    return get_records(type, status, publisher, year, start=start, limit=limit)
```

This is not detected in the specifications, but I did not find a way to explicitly document them. 

Regards
Luca

**Comment by greyli** [comment, marked as answer]
Hi, I'm glad you like it.

I would recommend defining these parameters as an input schema (with `query` location):

```python
class RecordQueryInput(Schema):
    type = fields.String(load_default='automatic')
    # ...

@bp.route("/records", methods=["GET"])
@input(RecordQueryInput, location='query')   # <-
@output(Record(many=True))
def get_records_from_form_data(query_data):
    # type = query_data['type']
    return get_records(**query_data)
```

Let me know if it works, not tested yet.

**Reply by lfoppiano** [reply]
Thanks! it works!

**Reply by lfoppiano** [reply]
@greyli one thing I just realised is that the "default" does not seems to work. I would expect these values to be always present in the `query_data` dictionary. However if i call the API without any parameter the `query_data` is empty. Is that normal?

**Reply by greyli** [reply]
Oops, the arg should be `load_default` instead of `default`.
